qcodes/instrument_drivers/Lakeshore/lakeshore_base.py

```diff
@@ -0,0 +1,302 @@
+                                          f'{{input_channel}}, '
+                                          f'{{powerup_enable}}, {{polarity}}, '
+                                          f'{{filter}}, {{delay}}',
+                                  get_cmd=f'OUTMODE? {output_index}')
+        if self._has_pid:
+            self.add_parameter('P', vals=validators.Numbers(0, 1000),
+                            get_parser=float, parameter_class=GroupParameter)
+            self.add_parameter('I', vals=validators.Numbers(0, 1000),
+                            get_parser=float, parameter_class=GroupParameter)
+            self.add_parameter('D', vals=validators.Numbers(0, 2500),
+                            get_parser=float, parameter_class=GroupParameter)
+            self.pid_group = Group([self.P, self.I, self.D],
+                                set_cmd=f"PID {output_index}, {{P}}, {{I}}, {{D}}",
+                                get_cmd=f'PID? {output_index}')
+
+        self.add_parameter('output_range',
+                           val_mapping=self.RANGES,
+                           set_cmd=f'RANGE {output_index}, {{}}',
+                           get_cmd=f'RANGE? {output_index}')
+
+        self.add_parameter('setpoint',
+                           vals=validators.Numbers(0, 400),
+                           get_parser=float,
+                           set_cmd=f'SETP {output_index}, {{}}',
+                           get_cmd=f'SETP? {output_index}')
+
+        # additional non Visa parameters
+        self.add_parameter('range_limits',
+                           set_cmd=None,
+                           vals=validators.Sequence(validators.Numbers(0,400),
+                                                    require_sorted=True,
+                                                    length=7),
+                           label='Temperature limits for ranges ', unit='K')
+
+        self.add_parameter('wait_cycle_time',
+                           set_cmd=None,
+                           vals=validators.Numbers(0,100),
+                           label='Time between two readings when waiting for'
+                                 'temperature to equilibrate', unit='s')
+        self.wait_cycle_time(0.1)
+
+        self.add_parameter('wait_tolerance',
+                           set_cmd=None,
+                           vals=validators.Numbers(0,100),
+                           label='Acceptable tolerance when waiting for '
+                                 'temperature to equilibrate', unit='')
+        self.wait_tolerance(0.1)
+
+        self.add_parameter('wait_equilibration_time',
+                           set_cmd=None,
+                           vals=validators.Numbers(0,100),
+                           label='Duration during which temperature has to be '
+                                 'within tolerance.', unit='')
+        self.wait_equilibration_time(0.5)
+
+
+        self.add_parameter('blocking_T',
```

Comment:
shouldn't this be a method?

Reply:
old Loop needs a parameter, so we end up with a parameter on top of a method instead of a method-only.